apps/fluent-tester/src/TestComponents/StrokeWidth/StrokeWidthTest.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { HOMEPAGE_STROKEWIDTH_TESTPAGE } from './consts';
+import { Test, TestSection, PlatformStatus } from '../Test';
+import { View, StyleSheet } from 'react-native';
+import { Stack } from '@fluentui-react-native/stack';
+import { stackStyle } from '../Common/styles';
+import { Text } from '@fluentui/react-native';
+import { createAppleTheme } from '@fluentui-react-native/apple-theme';
+
+const styles = StyleSheet.create({
+  root: {
+    paddingBottom: 16,
+  },
+  exampleStrokeBackground: {
+    height: 100,
+    backgroundColor: '#f0f0f0',
+    display: 'flex',
+    justifyContent: 'center',
+  },
+});
+
+interface StrokeWidthTestComponentProps {
+  name: string;
+  strokeWidth: number;
+}
+
+const StrokeWidthTestComponent: React.FunctionComponent<StrokeWidthTestComponentProps> = (props: StrokeWidthTestComponentProps) => {
+  const exampleStrokeStyle = { height: props.strokeWidth, backgroundColor: 'grey' };
+
+  return (
+    <View style={styles.root}>
+      <Text variant="bodySemibold">{props.name + ': ' + props.strokeWidth + 'pt'}</Text>
+      <View style={styles.exampleStrokeBackground}>
+        <View style={exampleStrokeStyle} />
+      </View>
+    </View>
+  );
+};
```

Comment:
Still trying to understand how to make sure components aren't getting rendered too many times. Is this component okay? Does exampleStrokeStyle need to be memoized?

Reply:
If exampleStrokeStyle  is taken outside of the component, it's fine. 

The way React components figure out if it needs to rerender is by shallow comparing all of its props and if anything is different, it will rerender the component. For objects (objects, arrays, functions), that means that if a new instance of that object is created and passed into the compare, then the shallow compare will fail because it's a different instance, even if the contents of the object are the same and so in reality no work needed to be done. You can think of it as a pointer comparison if you're used to native.

Anything inside of a (function) component is rerun when something asks components to rerender. So that means any instantiation of a const object will be redone, and new instances of those objects will be created. In this case `exampleStrokeStyle` has a new instance on every render and so the inner most view will be invalidated on every render.

Moving the const object outside the components works because that code is only run once.